lib/events.js

```diff
@@ -155,28 +311,52 @@ EventEmitter.prototype.emit = function emit(type) {
     needDomainExit = true;
   }
 
-  var isFn = typeof handler === 'function';
   len = arguments.length;
+
+  if (typeof handler !== 'function') {
+    onceCount = handler.onceCount;
+    useOnce = (onceCount !== 0);
+    isArray = (onceCount !== undefined);
+  }
+
   switch (len) {
     // fast cases
     case 1:
-      emitNone(handler, isFn, this);
+      if (!useOnce)
+        emitNone(handler, !isArray, this);
+      else
+        emitNoneOnce(handler, !isArray, this, type);
```

Comment:
Why push the `!isArray` branch into each `emit{None,One,Two,Three}{Once,}` function?